tests/unit/test_lua.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+
+from lupa._lupa import LuaRuntime
+from dredis.keyspace import Keyspace
+from dredis.lua import RedisScriptError
+from dredis.lua import LuaRunner, RedisLua
+
+
+def test_lua_return_redis_types_run():
+    k = Keyspace()
+    runner = LuaRunner(k)
+    lua_script = """return {'test', true, false, 10, 20.3, {4}}"""
+
+    assert runner.run(lua_script, [], []) == ['test', 1, None, 10, 20, [4]]
+
+
+def test_lua_table_with_error_run():
+    k = Keyspace()
+    runner = LuaRunner(k)
+    lua_script_err = """return {err='This is a ValueError'}"""
+
+    with pytest.raises(ValueError) as e:
+        runner.run(lua_script_err, [], [])
+
+    assert str(e.value) == 'This is a ValueError'
+
+
+def test_lua_table_with_ok_run():
+    k = Keyspace()
+    runner = LuaRunner(k)
+    lua_script_ok = """return {ok='Everything is OK'}"""
+
+    assert runner.run(lua_script_ok, [], []) == 'Everything is OK'
+
+
+def test_redislua_return_lua_types_call():
+    k = Keyspace()
+    lua_runtime = LuaRuntime(unpack_returned_tuples=True)
+    redis_lua = RedisLua(k, lua_runtime)
+    lua_script = """return {'test', true, false, 10, 20.3, {'another string'}, redis.call('ping')['ok']}"""
```

Comment:
this doesn't test that a table with `ok` *gets converted* to a string (you're doing the work inside of lua instead of outside). I'd change it to be:

```python
lua_script = """return {'test', true, false, 10, 20.3, {'another string'}, redis.call('ping')}"""
```

Then you're testing that the string from PING gets converted correctly (outside of lua).